diff --git a/node_modules/@greenwood/cli/src/config/rollup.config.js b/node_modules/@greenwood/cli/src/config/rollup.config.js
index 0894971..70175f0 100644
--- a/node_modules/@greenwood/cli/src/config/rollup.config.js
+++ b/node_modules/@greenwood/cli/src/config/rollup.config.js
@@ -7,7 +7,7 @@ import * as walk from 'acorn-walk';
 
 // https://github.com/rollup/rollup/issues/2121
 // would be nice to get rid of this
-function cleanRollupId(id) {
+function cleanRollupId(id = '') {
   return id.replace('\x00', '').replace('?commonjs-proxy', '');
 }
 
@@ -27,7 +27,7 @@ function greenwoodResourceLoader (compilation, browser = false) {
     async resolveId(id, importer, options) {
       const { userWorkspace, scratchDir } = compilation.context;
       const normalizedId = cleanRollupId(id);
-      const importerUrl = new URL(`file://${importer ?? ''}`);
+      const importerUrl = new URL(`file://${cleanRollupId(importer) ?? ''}`);
       const isUserWorkspaceImporter = importerUrl?.pathname?.startsWith(userWorkspace.pathname);
       const isScratchDirImporter = importerUrl?.pathname?.startsWith(scratchDir.pathname);
 
diff --git a/node_modules/@greenwood/cli/src/lib/walker-package-ranger.js b/node_modules/@greenwood/cli/src/lib/walker-package-ranger.js
index 5970487..4592882 100644
--- a/node_modules/@greenwood/cli/src/lib/walker-package-ranger.js
+++ b/node_modules/@greenwood/cli/src/lib/walker-package-ranger.js
@@ -1,9 +1,11 @@
 import fs from 'fs';
+import { isBuiltin } from 'node:module';
 
 // priority if from L -> R
 const SUPPORTED_EXPORT_CONDITIONS = ['import', 'module-sync', 'default'];
 const IMPORT_MAP_RESOLVED_PREFIX = '/~';
 const importMap = new Map();
+// TODO make diagnostics a Map
 const diagnostics = {};
 
 function updateImportMap(key, value, resolvedRoot) {
@@ -68,7 +70,9 @@ function derivePackageRoot(resolved) {
     root = root.substring(0, root.lastIndexOf(segment));
   }
 
-  return root;
+  return root !== ''
+    ? root
+    : null;
 }
 
 // helper function to convert export patterns to a regex (thanks ChatGPT :D)
@@ -235,6 +239,13 @@ async function walkPackageJson(packageJson = {}, walkedPackages = new Set()) {
 
             await walkPackageJson(resolvedPackageJson, walkedPackages);
           }
+        } else {
+          // ignore built-ins since NodeJS resolves them automatically
+          // https://github.com/nodejs/node/issues/56652
+          // https://nodejs.org/api/modules.html#built-in-modules
+          if (!isBuiltin(resolved)) {
+            diagnostics[dependency] = `WARNING: No package.json resolved for => \`${dependency}\`, resolved to \`${resolved}\``;
+          }
         }
       }
     }
