diff --git a/node_modules/@greenwood/cli/src/lib/walker-package-ranger.js b/node_modules/@greenwood/cli/src/lib/walker-package-ranger.js
index 5ce30c1..f4f4d0f 100644
--- a/node_modules/@greenwood/cli/src/lib/walker-package-ranger.js
+++ b/node_modules/@greenwood/cli/src/lib/walker-package-ranger.js
@@ -167,9 +167,15 @@ async function walkPackageJson(packageJson = {}) {
               ? esmPath
               : fallbackPath;
           } else if (exportMapEntry.import || exportMapEntry.default) {
-            packageExport = exportMapEntry.import
-              ? exportMapEntry.import
-              : exportMapEntry.default;
+            // TODO this is super slow for some reason?  not sure if all the ?
+            // On Node 20 (for .env), this seems faster / not an issue, but then ES Modules Shim is 404???
+            packageExport = exportMapEntry.import?.default?.default // https://unpkg.com/browse/tslib@2.6.2/package.json
+              ? exportMapEntry.import.default.default
+              : exportMapEntry.import?.default // https://unpkg.com/browse/lru-cache@10.1.0/package.json
+                ? exportMapEntry.import.default
+                : exportMapEntry.import
+                  ? exportMapEntry.import
+                  : exportMapEntry.default;
 
             // use the dependency itself as an entry in the importMap
             if (entry === '.') {
