diff --git a/node_modules/@greenwood/plugin-adapter-aws/src/index.js b/node_modules/@greenwood/plugin-adapter-aws/src/index.js
index 724fccf..69c10f2 100644
--- a/node_modules/@greenwood/plugin-adapter-aws/src/index.js
+++ b/node_modules/@greenwood/plugin-adapter-aws/src/index.js
@@ -33,7 +33,7 @@ function generateOutputFormat(id, type) {
         format = formData;
         delete headers['content-type'];
       } else if(contentType.includes('application/json')) {
-        format = JSON.stringify(body);
+        format = body;
       }
 
       const req = new Request(new URL(\`\${rawPath}\${queryParams}\`, \`http://\${headers.host}\`), {
